execInspect throws JsonMappingException

## Description

When trying to run a command in an already running docker container, trying to inspect the
exec throws a JsonMappingException, complaining about `exitCode` being null. My specific use case is starting
a program that shouldn't exit(except on error), which I thought could explain the exitcode being null(since it doesn't exit), but trying with `echo` or `ls` results in the same error.

## How to reproduce

```scala
      val exec = docker.execCreate(
        yourContainerId,
        Array( "ls")
      )
      docker.execStart(exec.id)

      try {
        val execState = docker.execInspect(exec.id)
        execState.running()
      } catch {
        case e : Throwable =>
          println("Exception: " + e.getCause)
      }
```

## What do you expect

The program is run and I can retrieve the result of running it.

## What happened instead

The program is run, but trying to retrieve the ExecState results in a JsonMappingException.

## Software:

- `docker version`: [Add the output of `docker version` here, both client and server]
```
Client:
 Version:      17.03.0-ce
 API version:  1.26
 Go version:   go1.8
 Git commit:   60ccb2265b
 Built:        Sun Mar  5 13:38:03 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.03.0-ce
 API version:  1.26 (minimum version 1.12)
 Go version:   go1.8
 Git commit:   60ccb2265b
 Built:        Sun Mar  5 13:38:03 2017
 OS/Arch:      linux/amd64
 Experimental: false
```
Also tried with version 1.12(updated to see if the error would go away)
- Spotify's docker-client version: 8.1.2

## Full backtrace

```text
play.api.http.HttpErrorHandlerExceptions$$anon$1: Execution exception[[DockerRequestException: Request error: GET unix://localhost:80/exec/98ac0c9b5330117e93200ba8a34e3d3d298b058b6a2e604c4cab521488ab7a4d/json: 200]]
	at play.api.http.HttpErrorHandlerExceptions$.throwableToUsefulException(HttpErrorHandler.scala:293)
	at play.api.http.DefaultHttpErrorHandler.onServerError(HttpErrorHandler.scala:220)
	at play.api.GlobalSettings$class.onError(GlobalSettings.scala:160)
	at play.api.DefaultGlobal$.onError(GlobalSettings.scala:188)
	at play.api.http.GlobalSettingsHttpErrorHandler.onServerError(HttpErrorHandler.scala:100)
	at play.core.server.netty.PlayRequestHandler$$anonfun$2$$anonfun$apply$1.applyOrElse(PlayRequestHandler.scala:100)
	at play.core.server.netty.PlayRequestHandler$$anonfun$2$$anonfun$apply$1.applyOrElse(PlayRequestHandler.scala:99)
	at scala.concurrent.Future$$anonfun$recoverWith$1.apply(Future.scala:346)
	at scala.concurrent.Future$$anonfun$recoverWith$1.apply(Future.scala:345)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:32)
Caused by: com.spotify.docker.client.exceptions.DockerRequestException: Request error: GET unix://localhost:80/exec/98ac0c9b5330117e93200ba8a34e3d3d298b058b6a2e604c4cab521488ab7a4d/json: 200
	at com.spotify.docker.client.DefaultDockerClient.propagate(DefaultDockerClient.java:2165)
	at com.spotify.docker.client.DefaultDockerClient.request(DefaultDockerClient.java:2104)
	at com.spotify.docker.client.DefaultDockerClient.execInspect(DefaultDockerClient.java:1815)
	at cerno.remote.types.DockerManager.startVpn(DockerManager.scala:117)
	at cerno.remote.types.DockerManager$$anonfun$startContainer$1.apply(DockerManager.scala:153)
	at cerno.remote.types.DockerManager$$anonfun$startContainer$1.apply(DockerManager.scala:150)
	at cerno.remote.types.DockerManager$$anonfun$1.apply$mcV$sp(DockerManager.scala:26)
	at akka.actor.Scheduler$$anon$2.run(Scheduler.scala:78)
	at akka.actor.LightArrayRevolverScheduler$$anon$2$$anon$1.run(LightArrayRevolverScheduler.scala:102)
	at scala.concurrent.impl.ExecutionContextImpl$AdaptedForkJoinTask.exec(ExecutionContextImpl.scala:121)
Caused by: javax.ws.rs.client.ResponseProcessingException: com.fasterxml.jackson.databind.JsonMappingException: Instantiation of [simple type, class com.spotify.docker.client.messages.ExecState] value failed (java.lang.NullPointerException): Null exitCode
 at [Source: org.glassfish.jersey.message.internal.ReaderInterceptorExecutor$UnCloseableInputStream@13b8f95; line: 1, column: 372]
	at org.glassfish.jersey.client.JerseyInvocation.translate(JerseyInvocation.java:806)
	at org.glassfish.jersey.client.JerseyInvocation.access$700(JerseyInvocation.java:92)
	at org.glassfish.jersey.client.JerseyInvocation$5.completed(JerseyInvocation.java:773)
	at org.glassfish.jersey.client.ClientRuntime.processResponse(ClientRuntime.java:198)
	at org.glassfish.jersey.client.ClientRuntime.access$300(ClientRuntime.java:79)
	at org.glassfish.jersey.client.ClientRuntime$2.run(ClientRuntime.java:180)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:271)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:267)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:315)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:297)
Caused by: com.fasterxml.jackson.databind.JsonMappingException: Instantiation of [simple type, class com.spotify.docker.client.messages.ExecState] value failed (java.lang.NullPointerException): Null exitCode
 at [Source: org.glassfish.jersey.message.internal.ReaderInterceptorExecutor$UnCloseableInputStream@13b8f95; line: 1, column: 372]
	at com.fasterxml.jackson.databind.JsonMappingException.from(JsonMappingException.java:263)
	at com.fasterxml.jackson.databind.deser.std.StdValueInstantiator.wrapAsJsonMappingException(StdValueInstantiator.java:445)
	at com.fasterxml.jackson.databind.deser.std.StdValueInstantiator.rewrapCtorProblem(StdValueInstantiator.java:464)
	at com.fasterxml.jackson.databind.deser.std.StdValueInstantiator.createFromObjectWith(StdValueInstantiator.java:258)
	at com.fasterxml.jackson.databind.deser.impl.PropertyBasedCreator.build(PropertyBasedCreator.java:135)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeUsingPropertyBased(BeanDeserializer.java:444)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1123)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:298)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:133)
	at com.fasterxml.jackson.databind.ObjectReader._bind(ObjectReader.java:1536)
Caused by: java.lang.NullPointerException: Null exitCode
	at com.spotify.docker.client.messages.AutoValue_ExecState.<init>(AutoValue_ExecState.java:40)
	at com.spotify.docker.client.messages.ExecState.create(ExecState.java:80)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.fasterxml.jackson.databind.introspect.AnnotatedMethod.call(AnnotatedMethod.java:117)
	at com.fasterxml.jackson.databind.deser.std.StdValueInstantiator.createFromObjectWith(StdValueInstantiator.java:256)
	at com.fasterxml.jackson.databind.deser.impl.PropertyBasedCreator.build(PropertyBasedCreator.java:135)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeUsingPropertyBased(BeanDeserializer.java:444)

```

--- a/src/main/java/com/spotify/docker/client/messages/ExecState.java
+++ b/src/main/java/com/spotify/docker/client/messages/ExecState.java
@@ -43,6 +43,7 @@ public abstract class ExecState {
   @JsonProperty("Running")
   public abstract Boolean running();
 
+  @Nullable
   @JsonProperty("ExitCode")
   public abstract Integer exitCode();
 

--- a/src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java
+++ b/src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java
@@ -3144,21 +3144,29 @@ public class DefaultDockerClientTest {
         containerId, new String[] {"sh", "-c", "exit 2"},
         createParams.toArray(new ExecCreateParam[createParams.size()]));
     final String execId = execCreation.id();
-
     log.info("execId = {}", execId);
+
+    final ExecState notStarted = sut.execInspect(execId);
+    assertThat(notStarted.id(), is(execId));
+    assertThat(notStarted.running(), is(false));
+    assertThat(notStarted.exitCode(), nullValue());
+    assertThat(notStarted.openStdin(), is(true));
+    assertThat(notStarted.openStderr(), is(true));
+    assertThat(notStarted.openStdout(), is(true));
+
     try (final LogStream stream = sut.execStart(execId)) {
       stream.readFully();
     }
 
-    final ExecState state = sut.execInspect(execId);
-    assertThat(state.id(), is(execId));
-    assertThat(state.running(), is(false));
-    assertThat(state.exitCode(), is(2));
-    assertThat(state.openStdin(), is(true));
-    assertThat(state.openStderr(), is(true));
-    assertThat(state.openStdout(), is(true));
+    final ExecState started = sut.execInspect(execId);
+    assertThat(started.id(), is(execId));
+    assertThat(started.running(), is(false));
+    assertThat(started.exitCode(), is(2));
+    assertThat(started.openStdin(), is(true));
+    assertThat(started.openStderr(), is(true));
+    assertThat(started.openStdout(), is(true));
 
-    final ProcessConfig processConfig = state.processConfig();
+    final ProcessConfig processConfig = started.processConfig();
     assertThat(processConfig.privileged(), is(false));
     if (execUserSupported) {
       assertThat(processConfig.user(), is("1000"));
@@ -3169,13 +3177,13 @@ public class DefaultDockerClientTest {
                Matchers.<List<String>>is(ImmutableList.of("-c", "exit 2")));
 
     if (dockerApiVersionLessThan("1.22")) {
-      final ContainerInfo containerInfo = state.container();
+      final ContainerInfo containerInfo = started.container();
       assertThat(containerInfo.path(), is("sh"));
       assertThat(containerInfo.args(),
                  Matchers.<List<String>>is(ImmutableList.of("-c", "while :; do sleep 1; done")));
       assertThat(containerInfo.config().image(), is(BUSYBOX_LATEST));
     } else {
-      assertNotNull(state.containerId(), "containerId");
+      assertNotNull(started.containerId(), "containerId");
     }
   }
 

--- a/src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java
+++ b/src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java
@@ -3149,7 +3149,11 @@ public class DefaultDockerClientTest {
     final ExecState notStarted = sut.execInspect(execId);
     assertThat(notStarted.id(), is(execId));
     assertThat(notStarted.running(), is(false));
-    assertThat(notStarted.exitCode(), nullValue());
+    if (dockerApiVersionLessThan("1.22")) {
+      assertThat(notStarted.exitCode(), is(0));
+    } else {
+      assertThat(notStarted.exitCode(), nullValue());
+    }
     assertThat(notStarted.openStdin(), is(true));
     assertThat(notStarted.openStderr(), is(true));
     assertThat(notStarted.openStdout(), is(true));

